shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -263,49 +368,73 @@ private void setSystemChromeSystemUIOverlayStyle(
-            break;
-        }
+    // You can't change the color of the status icons until SDK 23.
+    if (systemChromeStyle.statusBarIconBrightness != null && Build.VERSION.SDK_INT >= 23) {
+      switch (systemChromeStyle.statusBarIconBrightness) {
+        case DARK:
+          // View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
+          flags |= 0x2000;
+          break;
+        case LIGHT:
+          flags &= ~0x2000;
+          break;
       }
-      if (systemChromeStyle.statusBarColor != null) {
-        window.setStatusBarColor(systemChromeStyle.statusBarColor);
+    }
+    // You can't override the enforced contrast for a transparent status bar until SDK 29.
+    // This overrides the translucent scrim that may be placed behind the bar on SDK 29+ to ensure
+    // contrast is appropriate when using full screen layout modes like Edge to Edge.
+    if (!systemChromeStyle.systemStatusBarContrastEnforced && Build.VERSION.SDK_INT >= 29) {
+      window.setStatusBarContrastEnforced(systemChromeStyle.systemStatusBarContrastEnforced);
+    }
+
+    // SYSTEM NAVIGATION BAR --------------------------------------------------------------
+    // You can't change the color of the system navigation bar until SDK 21.
+    // If transparent, SDK 29 and higher may apply a translucent scrim behind 2/3 button navigation
+    // bars to ensure proper contrast. This can be overridden with
+    // SystemChromeStyle.systemNavigationBarContrastEnforced.
+    if (systemChromeStyle.systemNavigationBarColor != null
+        && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      window.setNavigationBarColor(systemChromeStyle.systemNavigationBarColor);
+    }
+    // You can't change the color of the navigation buttons until SDK 26.
+    if (systemChromeStyle.systemNavigationBarIconBrightness != null
+        && Build.VERSION.SDK_INT >= 26) {
+      switch (systemChromeStyle.systemNavigationBarIconBrightness) {
+        case DARK:
+          // View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
+          flags |= 0x10;
+          break;
+        case LIGHT:
+          flags &= ~0x10;
+          break;
       }
     }
+    // You can't change the color of the navigation bar divider color until SDK 28.
     if (systemChromeStyle.systemNavigationBarDividerColor != null && Build.VERSION.SDK_INT >= 28) {
       window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
       window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
       window.setNavigationBarDividerColor(systemChromeStyle.systemNavigationBarDividerColor);
     }
+
+    // You can't override the enforced contrast for a transparent navigation bar until SDK 29.
+    // This overrides the translucent scrim that may be placed behind 2/3 button navigation bars on
+    // SDK 29+ to ensure contrast is appropriate when using full screen layout modes like
+    // Edge to Edge.
+    if (!systemChromeStyle.systemNavigationBarContrastEnforced && Build.VERSION.SDK_INT >= 29) {
```

Comment:
nit: constant. although 29 is more readable than the constant 😅

Reply:
I think the constant failed when I used it, that's why I switched to the numbers.

Reply:
ah. That's probably due to the some jar that uses an older version. cc @dnfield do you know which one might be?  My understanding is that the junit tests currently use 30.

Reply:
I think it is the actual constants that are missing, it wasn't the tests that failed, just compilation.
AFAICT they are here: .cipd/pkgs/8/_current/systrace/catapult/devil/devil/android/sdk/version_codes.py

Reply:
I just pointed javadoc to 30. 

What's the failure you get when you use the constant?

Reply:
The build shoul dbe pointing to 30, based on https://github.com/flutter/buildroot/blob/fc82ca44f1ae85c5fbfa06bd313477539ca50729/build/config/android/config.gni#L89 and https://github.com/flutter/buildroot/blob/fc82ca44f1ae85c5fbfa06bd313477539ca50729/build/config/android/config.gni#L76 and https://github.com/flutter/buildroot/blob/fc82ca44f1ae85c5fbfa06bd313477539ca50729/build/config/android/config.gni#L18-L25

Reply:
was this a recent failure?

Reply:
Oh it looks like it's fine. I think I was using `Build.VERSION_CODES.OREO` instead of `Build.VERSION_CODES.O`. I did not check and see that they are truncating them now. :)

Reply:
Hmm, it looks like changing the the build version constants causes lint failures: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8843590413040618720/+/u/android_lint/dart_bin_main.dart/stdout

I'll revert back to the way it was.

Reply:
Those errors look like real errors, no? Are we missing API guards?

Reply:
Ahh I see the linter is failing to understand the constants.